codec/amino.go

```diff
@@ -23,23 +25,33 @@ func init() {
 	Cdc = cdc.Seal()
 }
 
-// Codec defines a type alias for an Amino codec.
-type Codec = amino.Codec
+// Codec defines a wrapper for an Amino codec that properly handles protobuf
+// types with Any's
+type Codec struct {
+	Amino *amino.Codec
+}
+
+var _ JSONMarshaler = &Codec{}
+
+func (cdc *Codec) Seal() *Codec {
+	cdc.Amino.Seal()
```

Comment:
Note, since we're working on a reference, there is no need to return the type at all.